prov/efa/test/efa_unit_test_cq.c

```diff
@@ -156,38 +158,83 @@ void test_cq_read_bad_send_status(struct efa_resource *resource, enum fi_ep_type
 	strerror = fi_cq_strerror(resource->cq, cq_err_entry.prov_errno, NULL, &err_buf, 0);
 	assert_int_equal(ret, 1);
 	assert_int_equal(cq_err_entry.err, FI_EIO);
-	assert_int_equal(cq_err_entry.prov_errno, FI_EFA_LOCAL_ERROR_UNRESP_REMOTE);
-	assert_string_equal(strerror, "Unresponsive receiver");
+	assert_int_equal(cq_err_entry.prov_errno, vendor_error);
+	assert_string_equal(strerror, expected_err);
 
 	efa_unit_test_buff_destruct(&send_buff);
 }
 
 /**
- * @brief test that RDM CQ's fi_cq_read()/fi_cq_readerr() works properly when rdma-core return bad status for send.
+ * @brief test that RDM CQ's fi_cq_read()/fi_cq_readerr() works properly when rdma-core returns
+ * unresponsive receiver error for send.
  *
  * When ibv_post_send() operation failed, fi_cq_read() should return -FI_EAVAIL, which means error available.
  * then user should call fi_cq_readerr() to get an error CQ entry that contain error code.
- * 
+ *
  * @param[in]	state		struct efa_resource that is managed by the framework
  */
-void test_ibv_cq_ex_read_bad_send_status(struct efa_resource **state)
+void test_ibv_cq_ex_read_bad_send_status_unresponsive_receiver(struct efa_resource **state)
 {
 	struct efa_resource *resource = *state;
-	test_cq_read_bad_send_status(resource, FI_EP_RDM);
+	test_cq_read_bad_send_status(resource, FI_EP_RDM, FI_EFA_LOCAL_ERROR_UNRESP_REMOTE,
```

Comment:
Tests which strcmp error messages seem a little brittle, but I guess there aren't a whole lot of alternatives to a test for errors.

Reply:
Yeah this test is to effectively cover the switch branches.